src/main/logutils.ts

```diff
@@ -110,48 +165,153 @@ const tailFile = (path: string) => {
+                }
+
+                if (value) {
+                    list_items.at(-1)?.push(value);
+                }
+
+                value = list_items.pop();
+                open_lists--;
+                continue;
+            }
+        }
+
+        value += c;
+    }
+
+    if (value) {
+        args_list.push(value)
+    }
+
+    if (open_lists > 0) {
+        throw `Unexpected EOL. There are ${open_lists} open tuples/lists.`
+    }
+
+    return new LogLine(timestamp, args_list)
+}
+
+    /**
  * Handle a line from the WoW log. 
  */
 const handleLogLine = (line: string) => {
-    if (line.includes("ARENA_MATCH_START")) {
-        handleArenaStartLine(line);
-    } else if (line.includes("ARENA_MATCH_END")) {
-        handleArenaStopLine(line);
-    } else if (line.includes("ENCOUNTER_START")) {
-        handleRaidStartLine(line);
-    } else if (line.includes("ENCOUNTER_END")) {
-        handleRaidStopLine(line);
-    } else if (line.includes("ZONE_CHANGE")) {
-        handleZoneChange(line);
-    } else if (line.includes("COMBATANT_INFO")) {
-        handleCombatantInfoLine(line);
-    } else if (line.includes("SPELL_AURA_APPLIED")){
-        handleSpellAuraAppliedLine(line);
+    // Parse line, only until the line token is encountered
+    let logLine = splitLogLine(line, 1)
+
+    // Check if we are interested in this event, and if not, discard it
+    if (interestingCombatLogEvents.indexOf(logLine.args[0]) === -1) {
+        return;
+    }
+
+    // Parse the full line
+    logLine = splitLogLine(line)
+
+    switch (logLine.args[0]) {
+        case "ARENA_MATCH_START":
+            handleArenaStartLine(logLine);
+            break;
+        case "ARENA_MATCH_END":
+            handleArenaStopLine(logLine);
+            break;
+        case "ENCOUNTER_START":
+            handleEncounterStartLine(logLine);
+            break;
+        case "ENCOUNTER_END":
+            handleEncounterStopLine(logLine);
+            break;
+        case "ZONE_CHANGE":
+            handleZoneChange(logLine);
+            break;
+        case "COMBATANT_INFO":
+            handleCombatantInfoLine(logLine);
+            break;
+        case "SPELL_AURA_APPLIED":
+            handleSpellAuraAppliedLine(logLine);
+            break;
+        default:
+            break;
     }
 }
 
 /**
  * Handle a line from the WoW log. 
  */
-const handleArenaStartLine = (line: string) => {
+const handleArenaStartLine = (line: LogLine) => {
     if (recorder.isRecording) return; 
-    const zoneID = parseInt(line.split(',')[1]);
-    const category = line.split(',')[3];
+    const zoneID = parseInt(line.args[1], 10);
 
     // If all goes to plan we don't need this but we do it incase the game
     // crashes etc. so we can still get a reasonable duration.
-    videoStartDate = getCombatLogDate(line);
+    videoStartDate = line.date();
 
     metadata = {
         name: "name",
-        category: category,
+        category: line.args[3],
```

Comment:
I think this is totally fine for now, but we could have a subclass that extends LogLine for each "interesting" event type with simple a .metadata() function to avoid having to have carefully set indexes in here.

Reply:
Oh, you've no idea how many things I wanted to do to this stuff, haha. I think that should go in a seperate PR because it'll be quite the change - but it'll come.